### PR TITLE
Allow callers to pass in a Storage implementation

### PR DESCRIPTION
**Pull request by Bestra**
This package contains really useful semantics for working with form fields, and I'd love it if callers could also provide their own `Storage` implementation when persisting or resuming. This would allow us to use, say, `localStorage` or any other arbitrary backend.

### CODE REVIEW

**Review by ashishkeshan — APPROVED** [review]
Nice job! Changes were very clear and nice test coverage too.

File: src/index.ts [hunk 1]
```diff
@@ -5,12 +5,13 @@ function shouldResumeField(field: HTMLInputElement | HTMLTextAreaElement): boole
   return !!field.id && field.value !== field.defaultValue && field.form !== submittedForm
 }
 
-type PersistOptions = {selector?: string; keyPrefix?: string}
+type PersistOptions = {selector?: string; keyPrefix?: string; storage?: Storage}
```
[the remark this answers is not in the document]
Comment (manuelpuyol) [reply src/index.ts:8]: I like the `Pick` idea :)

**Review by manuelpuyol — APPROVED** [review]

File: src/index.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (Bestra) [reply src/index.ts]: 👍 I've swapped in `Pick<>` for both arguments.

**Comment by Bestra** [comment]
@manuelpuyol do you think we could cut a release that includes this change sometime in the near future 🙇 ?

**Pull request merged**